great_expectations/data_context/types/base.py

```diff
@@ -281,23 +281,31 @@ def __init__(
         self,
         class_name,
         module_name=None,
+        credentials=None,
         assets=None,
         base_directory=None,
         glob_directive=None,
         default_regex=None,
         batch_identifiers=None,
-        credentials=None,
+        sorters=None,
+        batch_spec_passthrough=None,
+        # S3
+        boto3_options=None,
         bucket=None,
         prefix=None,
-        delimiter=None,
         max_keys=None,
-        boto3_options=None,
-        sorters=None,
-        batch_spec_passthrough=None,
+        # Azure
+        azure_options=None,
+        container=None,
+        name_starts_with=None,
+        # Both S3/Azure
+        delimiter=None,
```

Comment:
It is important to note that Azure uses different terminology to describe it's components.
* `bucket` is `container`
* `prefix` is `name_starts_with`

We keep these names consistent in our public API to ensure consistency between the SDK and our service. While it feels a bit duplicative and cumbersome to add each of these in various places around the codebase, the solution is not in the scope of this PR. We will have to develop child classes for each connector and provider at some point.

Reply:
Thank you for this note, and including the in-line comments for the parameters :)